src/ol/webgl/Helper.js

```diff
@@ -410,8 +410,8 @@ class WebGLHelper extends Disposable {
     const gl = this.getGL();
     const bufferKey = getUid(buf);
     const bufferCacheEntry = this.bufferCache_[bufferKey];
-    if (!gl.isContextLost()) {
-      gl.deleteBuffer(bufferCacheEntry.buffer);
+    if (bufferCacheEntry && !gl.isContextLost()) {
+      gl.deleteBuffer(bufferCacheEntry.webGlBuffer);
```

Comment:
The `buffer` property on the cache entry is an instance of a `WebGLArrayBuffer` from the `ol/webgl/Buffer.js` module.  The `webGlBuffer` property is an instance of a [`WebGLBuffer`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLBuffer).  The [`gl.deleteBuffer()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/deleteBuffer) method expects the former.